exonum-java-binding-core/rust/ejb-app/src/main.rs

```diff
@@ -1,12 +1,36 @@
+extern crate exonum_btc_anchoring;
 extern crate exonum_configuration;
 extern crate java_bindings;
+extern crate serde;
+#[macro_use]
+extern crate serde_derive;
 
+use exonum_btc_anchoring::ServiceFactory as BtcAnchoringServiceFactory;
 use exonum_configuration::ServiceFactory as ConfigurationServiceFactory;
+use java_bindings::exonum::helpers::config::ConfigFile;
 use java_bindings::exonum::helpers::fabric;
 
+const PATH_TO_SERVICES_TO_ENABLE: &str = "ejb_app_services.toml";
+const CONFIGURATION_SERVICE: &str = "configuration";
+const BTC_ANCHORING_SERVICE: &str = "btc-anchoring";
+
+#[derive(Serialize, Deserialize)]
+struct ServicesToEnable {
+    services: Vec<String>,
+}
+
 fn main() {
-    let builder = fabric::NodeBuilder::new()
-        .with_service(Box::new(java_bindings::JavaServiceFactory))
-        .with_service(Box::new(ConfigurationServiceFactory));
+    let services: ServicesToEnable = ConfigFile::load(PATH_TO_SERVICES_TO_ENABLE).unwrap();
+    let mut builder =
+        fabric::NodeBuilder::new().with_service(Box::new(java_bindings::JavaServiceFactory));
+    for service in &services.services {
+        if service == CONFIGURATION_SERVICE {
```

Comment:
Can we have a map of system service factories (name -> factory)? Then we can get rid of this chain of if/else if/…/else, which will grow a little:
```
for service_name in service_to_enable:
  if (system_services.contains(service_name)):
    builder = builder.with_service(system_services.get(service_name));
  else:
    panic("unknown service: " + service_name);
```

You can even put there a java service factory, and append its name to the list of service-to-enable.

Reply:
Okay, will try